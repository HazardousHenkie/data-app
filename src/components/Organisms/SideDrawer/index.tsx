import React, { useState } from 'react'

import Drawer, { DrawerProps } from '@material-ui/core/Drawer'
import IconButton from '@material-ui/core/IconButton'
import MenuIcon from '@material-ui/icons/Menu'

const SideDrawer: React.FC<Omit<DrawerProps, 'anchor' | 'onClose'>> = ({
    children
}) => {
    const [openDrawer, setOpenDrawer] = useState(false)

    const toggleDrawer = (open: boolean) => (
        event: React.KeyboardEvent | React.MouseEvent
    ) => {
        if (
            event.type === 'keydown' &&
            ((event as React.KeyboardEvent).key === 'Tab' ||
                (event as React.KeyboardEvent).key === 'Shift')
        ) {
            return
        }

        setOpenDrawer(open)
    }

    return (
        <>
            <IconButton
                color="inherit"
                aria-label="open drawer"
                onClick={toggleDrawer(true)}
                edge="start"
            >
                <MenuIcon />
            </IconButton>

            <Drawer
                anchor="left"
                open={openDrawer}
                onClose={toggleDrawer(false)}
            >
                {children}
            </Drawer>
        </>
    )
}

export default SideDrawer
