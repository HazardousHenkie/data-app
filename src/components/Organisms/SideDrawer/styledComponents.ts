import styled from 'styled-components'

import Drawer from '@material-ui/core/Drawer'

const StyledDrawer = styled(Drawer)`
    &&& {
        z-index: 1400 !important;
    }
`

export default StyledDrawer
