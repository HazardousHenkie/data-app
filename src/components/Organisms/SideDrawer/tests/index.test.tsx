import React from 'react'
import renderer from 'react-test-renderer'

import MockingAppComponent from 'utils/testing/MockingAppComponent'
import SideDrawer from '../index'

describe('<SideDrawer />', () => {
    it('should render like snapshot', () => {
        const component = renderer.create(
            <MockingAppComponent>
                <SideDrawer />
            </MockingAppComponent>
        )

        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
