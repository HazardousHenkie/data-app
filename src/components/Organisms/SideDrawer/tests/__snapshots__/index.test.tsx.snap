// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideDrawer /> should render like snapshot 1`] = `
<button
  aria-label="open drawer"
  className="MuiButtonBase-root MuiIconButton-root sc-AxiKw jWiChx MuiIconButton-colorInherit MuiIconButton-edgeStart"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </span>
</button>
`;
